content/_code-samples/airgapped-wallet/py/airgapped-wallet.py

```diff
@@ -0,0 +1,218 @@
+    # Scan the QR code and transmit the signed_tx blob to an online machine (Machine 2) to relay it to the XRPL
+    my_tx_payment_signed = safe_sign_transaction(transaction=my_tx_payment, wallet=_wallet)
+
+    img = qrcode.make(my_tx_payment_signed.to_dict())
+    img.save(get_path("/Wallet/transactionID.png"))
+    image = Image.open(get_path("/Wallet/transactionID.png"))
+    image.show()
+
+
+def get_path(file):
+    """
+    Get path (filesystem management)
+    """
+
+    global File_
+    # Checks what OS is being us
+    OS = platform.system()
+    usr = Path.home()
+
+    # Get PATH format based on the OS
+    if OS == "Windows":
+        File_ = PureWindowsPath(str(usr) + file)
+    if OS == "Linux":
+        File_ = PurePath(str(usr) + file)
+
+    return str(File_)
+
+
+def main():
+    global File, Path_
+
+    # Gets the machine's operating system (OS)
+    OS = platform.system()
+    usr = Path.home()
+    if OS == "Windows":
+        # If it's Windows, use this path:
+        File = PureWindowsPath(str(usr) + '/Wallet')
+        Path_ = str(PureWindowsPath(str(usr)))
+    if OS == "Linux":
+        # If it's Linux, use this path:
+        File = PurePath(str(usr) + '/Wallet')
+        Path_ = str(PurePath(str(usr)))
+
+    # If the Wallet's folder already exists, continue on
+    if os.path.exists(File) and os.path.exists(get_path("/Wallet/public.txt")):
+        while True:
+            ask = int(input("\n 1. Transact XRP"
+                            "\n 2. Generate an XRP wallet (read only)"
+                            "\n 3. Showcase XRP Wallet Address (QR Code)"
+                            "\n\n Enter Index: "
+                            ))
+
+            if ask == 1:
+                password = str(input("             Enter Password: "))
+                amount = float(input("\n           Enter XRP To Send: "))
+                destination = input("\n        Enter Destination: ")
+                wallet_sequence = int(input("\n    Enter Wallet Sequence: "))
+                ledger_sequence = int(input("Look up the latest ledger sequence on testnet.xrpl.org and enter it below!"
+                                            "\n    Enter Ledger Sequence: "))
+
+                sign_transaction(_xrp_amount=amount,
+                                 _destination=destination,
+                                 _ledger_seq=ledger_sequence,
+                                 _wallet_seq=wallet_sequence,
+                                 password=password
+                                 )
+
+                del destination, amount, wallet_sequence, ledger_sequence
+
+            if ask == 2:
+                _pub, _seed = create_wallet()
+
+            if ask == 3:
+                with open(get_path("/Wallet/public.txt"), "r") as f:
+                    print(f"\n  Wallet Address: {f.read()}")
+
+                image = Image.open(get_path("/Wallet/public.png"))
+                image.show()
+
```

Comment:
Allow for graceful exit (Part 2/2)
```suggestion
if ask == 4:
       break
```